src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -302,7 +299,7 @@ private TestWorkspace CreateTestWorkspace(string[] markups, out Dictionary<strin
                 _ => throw new ArgumentException($"language name {languageName} is not valid for a test workspace"),
             };
 
-            SetSolutionProviderWorkspace(workspace);
+            RegisterWorkspaceForLsp(workspace);
```

Comment:
I wonder if instead of this and the TestLspWorkspaceRegistrationService if we just used the normal workspace registration service and had the TestWorkspace always call the register method on itself when it's created

Reply:
I'm sure I tried this at some point, but can't remember why I didn't go with it. I think my brain reset on Jan 1 though, so I'll give it another go :)

Reply:
Okay, this doesn't work, I think because of layering given that the "normal" workspace registration service is in the Visual Studio layer. We'd have to move it down, but there was pretty strong feedback against that. Could always do a follow up.